compiler/rustc_expand/src/mbe/macro_rules.rs

```diff
@@ -628,6 +628,44 @@ fn check_lhs_nt_follows(sess: &ParseSess, def: &ast::Item, lhs: &mbe::TokenTree)
     // after parsing/expansion. we can report every error in every macro this way.
 }
 
+fn is_empty_token_tree(sess: &ParseSess, seq: &mbe::SequenceRepetition) -> bool {
+    if seq.separator.is_some() {
+        return false;
+    }
+    use mbe::TokenTree;
+    let mut index = 0;
```

Comment:
Instead of an index you could use a (peekable) iterator instead, and `while let Some(tt) = it.next()`.